app/client/templates/elements/compileContract.js

```diff
@@ -21,6 +21,8 @@ Template['elements_compileContract'].onCreated(function() {
     TemplateVar.set('value', '');
     TemplateVar.set('constructorInputs', []);
     TemplateVar.set('selectedType', this.data.onlyByteCode ? 'byte-code' : 'source-code');
+    TemplateVar.set('compiledContracts', JSON.parse(localStorage['compiledContracts'])  || null);
+    TemplateVar.set('selectedContract', JSON.parse(localStorage['selectedContract'])  || null);
```

Comment:
JSON.parse will fail if localStorage value is undefined. It treats `undefined` as string, failing with "Uncaught SyntaxError: Unexpected token u in JSON at position 0".

To accomplish that you can use the `|| null` inside the JSON.parse statement: 
`JSON.parse(localStorage['selectedContract']  || null)`

demo here: http://codepen.io/evertonfraga/pen/LZRaby